doc/compiler/optimizer/IntroReadLogFile.md

```diff
@@ -0,0 +1,1188 @@
+ [0x7fef12cec7d0]	mov	rdi, rax		# MOV8RegReg
+ [0x7fef12cec700]	mov	rax, qword ptr [vfp]		# L8RegMem, #SPILL8, SymRef  <#SPILL8_1122 0x7fef12cabf70>[#1122  Auto] [flags 0x80000000 0x0 ]#, spilled for acall
+```
+Register spill around a call at `0x7fef12bbed50`. The spills are added by the register assigner. Sometimes it cannot
+satisfy all the values that need to be kept alive at a particular program point given the constraints of an architecture.
+Sometimes there are more virtual registers than the number of real registers that are available. It will spill some of
+the values into the memory stack slots. Once in a better state, after the call is done, the values are restored from
+the memory into the register. Certain registers are set aside as being preserved before the call.
+
+One of the things on debugging register allocator problem is to look at if there is a lot of spilling. If there is,
+exam closely and see if there is an issue of save and restore registers. One possibility could be that the restore is
+not done correctly on one of the paths, for example missing a restore, or restoring a wrong register, or a wrong value
+is used.
+
+### 7.7 Post Binary Instructions
+
+```
+                 +--------------------------------------- instruction address
+                 |        +----------------------------------------- instruction offset from start of method
+                 |        |                   +------------------------------------------ corresponding TR::Instruction instance
+                 |        |                   |  +-------------------------------------------------- code bytes
+                 |        |                   |  |                          +-------------------------------------- opcode and operands
+                 |        |                   |  |                          |				+----------- additional information
+                 |        |                   |  |                          |				|
+                 V        V                   V  V                          V				V
+0x7fef13a746d0 ffffffd0 [0x7fef12ad0bf0]                                    Label L0084:			# label
+0x7fef13a746d0 ffffffd0 [0x7fef12ad0d80] 48 bf 78 2c 34 00 00 00 00 00      mov	rdi, 0x0000000000342c78	# MOV8RegImm64
+
+0x7fef13a74700 00000000 [0x7fef12d881e0] 48 8b 44 24 18                     mov	rax, qword ptr [rsp+0x18]		# L8RegMem, SymRef [#1164 +24]
+```
+
+`"Post Binary Instructions"` is the last thing in the JIT compilation.
+
+`0x7fef13a74700`: the code cache address or instruction address where the bytecode instruction is.
+
+`00000000`: instruction offset from start of method.
+
+`0x7fef12ad1d50`: TR instruction address.
+
+`48 8b 44 24 18`: bytes that represents the instruction.
+
+Code gen also generates prologue and epilogue. The linkage convention requires to save certain values on the way into
```

Comment:
We should update the following sentence to cover both prologue and epilogue operations

> The linkage convention requires to save certain values on the way into a call, set up the stack pointer, grow the stack for a frame, check stack overflow etc.

Maybe something like:

Linkage conventions may require managements of incoming parameters and return values, adjustments of stack pointer to allocate/deallocate a call frame, saving / restoring preserved register values, checks for stack overflow, etc.

Reply:
Addressed in 9c5fd3b